pkg/runtime/framework/plugins/coscheduling/indexer_test.go

```diff
@@ -0,0 +1,398 @@
+									Template: batchv1.JobTemplateSpec{
+										Spec: batchv1.JobSpec{
+											Template: corev1.PodTemplateSpec{
+												Spec: corev1.PodSpec{
+													RuntimeClassName: ptr.To(constants.DatasetInitializer),
+												},
+											},
+										},
+									},
+								},
+								{
+									Name: constants.ModelInitializer,
+									Template: batchv1.JobTemplateSpec{
+										Spec: batchv1.JobSpec{
+											Template: corev1.PodTemplateSpec{
+												Spec: corev1.PodSpec{
+													RuntimeClassName: ptr.To(constants.ModelInitializer),
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			want: []string{constants.DatasetInitializer, constants.ModelInitializer},
+		},
+		"TrainingRuntime with one ReplicatedJob and RuntimeClassName set": {
+			obj: &trainer.TrainingRuntime{
+				ObjectMeta: metav1.ObjectMeta{Namespace: metav1.NamespaceDefault, Name: "test"},
+				Spec: trainer.TrainingRuntimeSpec{
+					Template: trainer.JobSetTemplateSpec{
+						Spec: jobsetv1alpha2.JobSetSpec{
+							ReplicatedJobs: []jobsetv1alpha2.ReplicatedJob{
+								{
+									Name: constants.ModelInitializer,
+									Template: batchv1.JobTemplateSpec{
+										Spec: batchv1.JobSpec{
+											Template: corev1.PodTemplateSpec{
+												Spec: corev1.PodSpec{
+													RuntimeClassName: ptr.To(constants.ModelInitializer),
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			want: []string{constants.ModelInitializer},
+		},
+		"TrainingRuntime with ReplicatedJobs where some RuntimeClassName are set and others are nil": {
+			obj: &trainer.TrainingRuntime{
+				ObjectMeta: metav1.ObjectMeta{Namespace: metav1.NamespaceDefault, Name: "test"},
+				Spec: trainer.TrainingRuntimeSpec{
+					Template: trainer.JobSetTemplateSpec{
+						Spec: jobsetv1alpha2.JobSetSpec{
+							ReplicatedJobs: []jobsetv1alpha2.ReplicatedJob{
+								{
+									Name: constants.DatasetInitializer,
+									Template: batchv1.JobTemplateSpec{
+										Spec: batchv1.JobSpec{
+											Template: corev1.PodTemplateSpec{
+												Spec: corev1.PodSpec{
+													RuntimeClassName: ptr.To(constants.DatasetInitializer),
+												},
+											},
+										},
+									},
+								},
+								{
+									Name: constants.ModelInitializer,
+									Template: batchv1.JobTemplateSpec{
+										Spec: batchv1.JobSpec{
+											Template: corev1.PodTemplateSpec{
+												Spec: corev1.PodSpec{
+													RuntimeClassName: nil,
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			want: []string{constants.DatasetInitializer},
+		},
+	}
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			got := IndexTrainingRuntimeContainerRuntimeClass(tc.obj)
+			if diff := cmp.Diff(tc.want, got, cmpopts.EquateEmpty()); len(diff) != 0 {
```

Comment:
```suggestion
			if diff := cmp.Diff(tc.want, got); len(diff) != 0 {
```

We should validate those more statically.